File: backend/src/main/java/com/minh/jewerlystore/service/CartService.java
package com.minh.jewerlystore.service;

import com.minh.jewerlystore.entity.User;
import com.minh.jewerlystore.payload.response.CartResponse;

public interface CartService {
    CartResponse getCart(User user);
    CartResponse addToCart(User user, Long productId, Integer quantity);
    CartResponse updateCartItemQuantity(User user, Long productId, Integer quantity);
    CartResponse removeFromCart(User user, Long productId);
    void clearCart(User user);
} 

File: frontend/src/services/cartService.js
import api from './api';

export const cartService = {
  getCart: async () => {
    const response = await api.get('/cart');
    return response.data;
  },

  addToCart: async (productId, quantity) => {
    const response = await api.post(`/cart/add/${productId}?quantity=${quantity}`);
    return response.data;
  },

  updateCartItem: async (productId, quantity) => {
    const response = await api.put(`/cart/update/${productId}?quantity=${quantity}`);
    return response.data;
  },

  removeFromCart: async (productId) => {
    const response = await api.delete(`/cart/remove/${productId}`);
    return response.data;
  },

  clearCart: async () => {
    const response = await api.delete('/cart/clear');
    return response.data;
  }
}; 